File: repository/src/main/java/com/SReMake/repository/system/impl/CasbinRuleRepositoryImpl.java
package com.SReMake.repository.system.impl;

//import com.SReMake.model.Tables;
//import com.SReMake.model.system.CasbinRule;
//import com.SReMake.repository.system.CasbinRuleRepository;
//import org.babyfish.jimmer.spring.repo.support.AbstractJavaRepository;
//import org.babyfish.jimmer.sql.JSqlClient;
//import org.springframework.stereotype.Repository;
//
//import java.util.Arrays;
//import java.util.List;
//
//@Repository
//public class CasbinRuleRepositoryImpl extends AbstractJavaRepository<CasbinRule, Integer> implements CasbinRuleRepository {
//
//    public CasbinRuleRepositoryImpl(JSqlClient sql) {
//        super(sql);
//    }
//
//    @Override
//    public long countMatches(String router) {
//        return this.sql.createQuery(Tables.CASBIN_RULE_TABLE).where(Tables.CASBIN_RULE_TABLE.v1().eq(router)).select(Tables.CASBIN_RULE_TABLE.count()).fetchOptional().orElse(0L);
//    }
//
//    @Override
//    public List<String> listMatchesRole(String... roles) {
//        return this.sql.createQuery(Tables.CASBIN_RULE_TABLE).where(Tables.CASBIN_RULE_TABLE.v0().in(Arrays.stream(roles).toList())).select(Tables.CASBIN_RULE_TABLE.v1()).fetchOptional().stream().toList();
//    }
//}

File: repository/src/main/kotlin/com/SReMake/repository/system/impl/CasbinRuleRepositoryImpl.kt
package com.SReMake.repository.system.impl

import com.SReMake.model.system.CasbinRule
import com.SReMake.model.system.v0
import com.SReMake.model.system.v1
import com.SReMake.repository.system.CasbinRuleRepository
import org.babyfish.jimmer.spring.repo.support.AbstractKotlinRepository
import org.babyfish.jimmer.sql.kt.KSqlClient
import org.babyfish.jimmer.sql.kt.ast.expression.count
import org.babyfish.jimmer.sql.kt.ast.expression.eq
import org.babyfish.jimmer.sql.kt.ast.expression.valueIn
import org.springframework.stereotype.Repository

@Repository
class CasbinRuleRepositoryImpl(sql: KSqlClient) : CasbinRuleRepository,
    AbstractKotlinRepository<CasbinRule, Int>(sql) {
    override fun countMatches(router: String) =
        this.sql.createQuery(CasbinRule::class) {
            where(table.v1.eq(router))
            select(count(table))
        }.fetchOne()


    override fun listMatchesRole(roles: Collection<String>): List<String> =
        this.sql.createQuery(CasbinRule::class) {
            where(table.v0.valueIn(roles))
            select(table.v1)
        }.execute().mapNotNull { it }

}